src/widgets/food_select.rs

```diff
@@ -83,14 +83,18 @@ impl<'a> Widget for FoodSelect<'a> {
 
                 ui.horizontal(|ui| {
                     ui.label("Search:");
-                    ui.text_edit_singleline(&mut search_text);
+                    if ui.text_edit_singleline(&mut search_text).changed() {
+                        search_text.truncate(search_text.trim_end_matches('\0').len());
```

Comment:
If `trim_end_matches` trims from the beginning of the string, then this line is wrong. Although that case will probably never occur, it is better to rewrite this as
```rust
search_text = search_text.trim_end_matches('\0').to_string();
```

Reply:
Ah I just realized that `trim_end_matches` only trims from the end :smiley:
Still, I prefer the suggested version for readability.

Reply:
Even better, you could use
```rust
search_text.remove_matches('/0');
```

Reply:
I used `trim_end_matches` since the null bytes are only at the end of the string when pasting from the game and `replace` would go through all characters instead of just the ones at the end until one doesn't match.
The use of `truncate` is because I wanted to make absolutely sure that no new `String` is created, maybe it is not needed.
But I can definitely change it.

Reply:
The performance hit from creating a new string is negligible, since this part is only execute once after the input changes. That's why I would prefer simplicity and readability over performance in this case.